{{cookiecutter.project_dash_slug}}/Dockerfile

```diff
@@ -0,0 +1,42 @@
+# Base Image
+FROM python:3.6.6-alpine3.7
+
+# Best practice: if you build contains several layers you should order them
+# from the less frequently changed to the more frequently changed (caching)
+
+# Install tools required for the project
+# Best practice:
+# 1. do not install unnecessary packages, keep it small.
+# 2. keep an alphabetic order to avoid duplicates
+RUN apk add --no-cache --update \
+    build-base \
+    git \
+    libffi-dev \
+    linux-headers \
+    pcre-dev \
+    postgresql-client \
+    python3-dev
+
+# Run as non root user (best practice)
+ENV USERNAME=syapse
+ENV PROJECT_NAME={{cookiecutter.project_dash_slug}}
+WORKDIR /home/$USERNAME/$PROJECT_NAME
+COPY . .
+RUN addgroup -g 303 -S $USERNAME \
+    && adduser -u 303 -S -G $USERNAME -h /home/$USERNAME $USERNAME
+RUN chown -R $USERNAME:$USERNAME /home/$USERNAME/$PROJECT_NAME
+USER $USERNAME
+
+# Install or update system dependencies
+ENV PIP_NO_CACHE_DIR=off
```

Comment:
Here are a few other helpful environment variables for Python and Pipenv:

```Dockerfile
##
# Some helpful environment variables:
#
#   - PYTHONFAULTHANDLER enables the fault handler: 
#       -> https://docs.python.org/3/library/faulthandler.html
#   - PYTHONUNBUFFERED ensures that you're getting realtime output when you tail the logs in the container: 
#       -> https://docs.python.org/3/using/cmdline.html#cmdoption-u
#   - PIP_DISABLE_PIP_VERSION_CHECK suppresses the warning message about outdated pip (useful since you're upgrading it on build here).
#   - PIPENV_COLORBLIND disables colored output in pipenv (it plays nicer with CI)
#   - PIPENV_NOSPIN disables the spinner (again, plays nicer with simpler terminals)
ENV PYTHONFAULTHANDLER=1 \
    PYTHONUNBUFFERED=1 \
    PIP_DISABLE_PIP_VERSION_CHECK=on \
    PIPENV_COLORBLIND=true \
    PIPENV_NOSPIN=true
```